### ISSUE TITLE
Coefficients reported by the tidy and coef method in a rule_fit model are not equal

### ISSUE THREAD

**Issue by frequena** [tidymodels/rules#26]
Hello! I am not sure whether this is the expected behavior, but when I use the function tidy() in a rules::rule_fit model, the list of features  (and their coefficients) reported do not coincide with the coef() output. 

If this is the correct behavior, which coefficients are being reported?

You can find an example below:

``` r
library(rules)
#> Le chargement a nécessité le package : parsnip
library(yardstick)
#> For binary classification, the first factor level is assumed to be the event.
#> Use the argument `event_level = "second"` to alter this as needed.
library(parsnip)
library(broom)
library(dplyr)
#> 
#> Attachement du package : 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
test_rulefit <- rules::rule_fit() %>%
  set_mode('classification') %>%
  set_engine('xrf') %>%
  fit(am ~ ., mtcars %>% mutate(am = as.factor(am)))
#> New names:
#> * . -> ....1
#> * . -> ....2
#> * . -> ....3
#> * . -> ....4
#> * . -> ....5
#> * ...

test_rulefit %>% tidy()
#> # A tibble: 5 x 3
#>   rule_id     rule                                          estimate
#>   <chr>       <chr>                                            <dbl>
#> 1 (Intercept) ( TRUE )                                        -5.01 
#> 2 gear        ( gear )                                         1.11 
#> 3 r0_3        ( qsec <  19.1849995 ) & ( wt <  3.18000007 )    0.162
#> 4 r1_3        ( hp <  222.5 ) & ( wt >= 3.01250005 )          -0.841
#> 5 r2_3        ( gear >= 3.5 ) & ( wt <  3.01250005 )           2.06

test_rulefit$fit %>% coef() %>% filter(coefficient_lambda.min != 0)
#>   coefficient_lambda.min        term                      rule
#> 1           -15.48490246 (Intercept)                      <NA>
#> 2            -2.30560971        r1_3  hp<222.500 & wt>=  3.013
#> 3            -1.31716484        r8_4 qsec>=17.410 & wt>= 3.013
#> 4             0.08077928        r7_3  gear>= 3.50 & qsec<18.76
#> 5             0.24635541        r0_3     qsec<19.18 & wt< 3.18
#> 6             0.50798063        r5_3  gear>= 3.50 & qsec<17.66
#> 7             3.71180516        gear                      <NA>
#> 8             5.16985334        r2_3    gear>=3.500 & wt<3.013
```

<sup>Created on 2021-03-19 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by simonpcouch** [comment]
Apologies for the delay here! Thanks for pointing this out.

The difference you're seeing here are related to how differences in how rules and xrf (a package that rules wraps) retrieve the `coef`ficients from the glmnet model. The two packages set the `s` argument to `coef.cv.glmnet()` differently.

The xrf package defaults to `'lambda.min'`:

https://github.com/holub008/xrf/blob/c59736a/R/xrf.R#L564-L566

rules defaults to setting `s` to the penalty to the value initially passed to `xrf` (default in `rule_fit()` is `0.1`):

https://github.com/tidymodels/rules/blob/3976a3d81d97864aec806b7de923fd778a9532ba/R/tidy_xrf.R#L44-L54

To get the similar results from the two packages, you can tell `xrf` which lambda you'd like:

``` r
library(rules)
#> Loading required package: parsnip
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union

test_rulefit <- rule_fit() %>%
  set_mode('classification') %>%
  set_engine('xrf') %>%
  fit(am ~ ., mtcars %>% mutate(am = as.factor(am)))

test_rulefit %>% 
  tidy()
#> # A tibble: 5 × 3
#>   rule_id     rule                                          estimate
#>   <chr>       <chr>                                            <dbl>
#> 1 (Intercept) ( TRUE )                                        -5.01 
#> 2 gear        ( gear )                                         1.11 
#> 3 r0_3        ( qsec <  19.1849995 ) & ( wt <  3.18000007 )    0.162
#> 4 r1_3        ( hp <  222.5 ) & ( wt >= 3.01250005 )          -0.841
#> 5 r2_3        ( gear >= 3.5 ) & ( wt <  3.01250005 )           2.06

test_rulefit$fit %>% 
  coef(lambda = .1) %>% 
  filter(coefficient_0.1 != 0)
#>   coefficient_0.1        term                     rule
#> 1      -5.0145751 (Intercept)                     <NA>
#> 2      -0.8407512        r1_3 hp<222.500 & wt>=  3.013
#> 3       0.1615640        r0_3    qsec<19.18 & wt< 3.18
#> 4       1.1096864        gear                     <NA>
#> 5       2.0604145        r2_3   gear>=3.500 & wt<3.013
```

<sup>Created on 2022-06-22 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Thanks again. :)
